### ISSUE TITLE
Add completed method in Pool

### ISSUE THREAD

**Issue by RewriteH** [josdejong/workerpool#354]
Instead of doing every time this:
```
const pool = workerpool.pool(
    path.resolve(__dirname, 'worker')
)
const tasks: workerpool.Promise<any, Error>[] = []

for (const entity of entities) {
 const task = pool.exec('worker', [entity])
 tasks.push(task)
}

await Promise.all(tasks)
await pool.terminate()
```

We can do:

```
const pool = workerpool.pool(
    path.resolve(__dirname, 'worker')
)

for (const entity of entities) {
 pool.exec('worker', [entity])
}

/* Wait all tasks to complete */ 
await pool.completed()
await pool.terminate()
```

It would be very handy, if this method is added in the future
For example, threads.js already have [this feature](https://threads.js.org/usage-pool#waiting-for-tasks-to-complete)

**Comment by josdejong** [comment]
Thanks for your suggestion. Actually, what you want is already there: just calling `await pool.terminate()` will by default gracefully shutdown the workerpool. It finishes all running tasks and does not accept new tasks.

From the docs:

> `Pool.terminate([force: boolean [, timeout: number]])`
>
>  If parameter `force` is false (default), workers will finish the tasks they are working on before terminating themselves. Any pending tasks will be rejected with an error 'Pool terminated'. When `force` is true, all workers are terminated immediately without finishing running tasks. If `timeout` is provided, worker will be forced to terminate when the timeout expires and the worker has not finished.


Docs: https://github.com/josdejong/workerpool#api

**Comment by josdejong** [comment]
Ooo wait a second, of course: `terminate` only gracefully finishes all _running_ tasks, but it will cancel and not execute _queued_ tasks. Sorry, my bad. 

So, yes, your suggestion makes sense, a method like `await pool.completed()`. Anyone interested in thinking this through and implementing such a method?

**Comment by Darshan-upadhyay1110** [comment]
`await pool.completed().then(result=>{
pool.terminate();
})`
is it the way we can do this ?

**Comment by josdejong** [comment]
Yes, you can keep track on all tasks and wait till they are all resolved, like the first example in https://github.com/josdejong/workerpool/issues/354#issue-1365597358. It's a bit cumbersome but not a showstopper.
